### ISSUE TITLE
Missing `range` method

### ISSUE THREAD

**Issue by ParadaCarleton** [JuliaLang/julia#43194]
1.6 documentation describes `range` as having signature:
```
    range(start[, stop]; length, stop, step=1)
```

So `range(1, 10)` should equal `range(1, 10; step=1)`. Instead:
```
    julia> range(1, 10)
ERROR: ArgumentError: At least one of `length` or `step` must be specified

julia> range(1, 10; step=1)
1:1:10
```

**Comment by oscardssmith** [comment]
this works as of 1.7. I think the docs just got updated too early.

**Comment by adityagi02** [comment]
I want to do this, can you please guide me @oscardssmith @ParadaCarleton

**Comment by oscardssmith** [comment]
TLDR is
1. find the docs.
2. click the edit on github button
3. add a compat annotation (see https://github.com/JuliaLang/julia/blob/35f0c911f443ae0d3933e1bce4321afd0f345aed/base/threadingconstructs.jl#L99-L117 for an example)
4. Make a PR with it (and maybe @ me)

**Comment by adityagi02** [comment]
This is the current picture of this:
https://github.com/JuliaLang/julia/blob/35f0c911f443ae0d3933e1bce4321afd0f345aed/base/range.jl#L49-L90

And this is after the changes, 
https://github.com/adityagi02/julia/blob/3766ddfeed7d3191b2a4a95979e6b1927d99c620/base/range.jl#L49-L92

@oscardssmith  please review & tell me the next step to proceed.

**Comment by oscardssmith** [comment]
Go to https://github.com/adityagi02/julia/tree/3766ddfeed7d3191b2a4a95979e6b1927d99c620, and you should see a button to make a PR against Base.

**Comment by fredrikekre** [comment]
Fixed with 1.7 doc deploy: https://docs.julialang.org/en/v1/base/math/#Base.range
